rudder-core/src/main/scala/com/normation/rudder/services/policies/ExpectedReportsUpdate.scala

```diff
@@ -132,7 +148,122 @@ class ExpectedReportsUpdateImpl(
     }
   }
 
-  /**
+
+
+  def saveNodeExpectedReports(configs: List[NodeExpectedReports]): Box[List[NodeExpectedReports]] = {
+    import Doobie._
+    import doobie.xa
+
+    configs.map(_.nodeId).toNel match {
+      case None          => Full(Nil)
+      case Some(nodeIds) =>
+
+        implicit val nodeIdsParam = Param.many(nodeIds)
+
+        type A = \/[(NodeId, NodeConfigId, DateTime), NodeExpectedReports]
+        val getConfigs: \/[Throwable, List[A]] = sql"""
+                           select nodeid, nodeconfigid, begindate, enddate, configuration
+                           from nodeconfigurations
+                           where nodeid in (${nodeIds: nodeIds.type}) and enddate is NULL
+                        """.query[A].list.attempt.transact(xa).run
+
+        type B = (NodeId, Vector[NodeConfigIdInfo])
+        val getInfos: \/[Throwable, List[B]] = sql"""
+                            select node_id, config_ids from nodes_info where node_id in (${nodeIds: nodeIds.type})
+                          """.query[B].list.attempt.transact(xa).run
+
+        // common part: find old configs and node config info for all config to update
+        val time_0 = System.currentTimeMillis
+        for {
+          oldConfigs  <- getConfigs
+          time_1      =  System.currentTimeMillis
+          _           =  TimingDebugLogger.debug(s"saveNodeExpectedReports: find old nodes config in ${time_1-time_0}ms")
+          configInfos <- getInfos
+          time_2      =  System.currentTimeMillis
+          _           =  TimingDebugLogger.debug(s"saveNodeExpectedReports: find old config info ${time_2-time_1}ms")
+          updated     <- doUpdateNodeExpectedReports(configs, oldConfigs, configInfos.toMap)
+          time_3      =  System.currentTimeMillis
+          _           =  TimingDebugLogger.debug(s"saveNodeExpectedReports: save configs etc in ${time_3-time_2}ms")
+        } yield {
+          configs
+        }
+    }
+  }
+
+  /*
+   * Save a nodeExpectedReport, closing the previous one is
+   * opened for that node (and a different nodeConfigId)
+   */
+  def doUpdateNodeExpectedReports(
```

Comment:
we used to have an other "update...", and it was before names settle. I will do a cleaning pass in a next release that will handle that and other "NoButReallyNowDoTheActuallyInterestingThing" names :)